### DISCUSSION TITLE
Messages are working on one path, but on the other path ([slug]) it throws No messages were configured on the provider.

### DISCUSSION THREAD

**Discussion by murbanowicz** [amannn/next-intl#1585] [Q&A]
Hi,
I am pulling hair out of my head for hours now and I have no idea.

I have so far pretty simple web - landing which is redirected to BLOG index (latest posts) and this is working - I get the translations there.
However, I have also /[locale]/blog/[slug] and for some weird reason, the translations are not working there throwing the mentioned error: `No messages were configured on the provider.` 

Both pages are using same layout (app router). I am on the latest version of the library.

Only last clue I have is that it may be related to the matcher config:
```

export const config = {
  // Match only internationalized pathnames
  matcher: [
    "/((?!api|static|studio|.*\\..*|_next).*)",
    "/",
    // `/(${routing.locales.join("|")})/:path*`,
  ],
};
```
however I had to comment out as you see, because otherwise the assets stop working, but I don't believe this is it.

Other clue can be that only on the post page (the one which is not working) I see another error about missing timezone, while on the posts list (which is working) I do not see this error.

Any idea, anything specific to double check etc?

I love the library and didn't have anything like that before.

**Comment by amannn** [comment]
Hmm, are you trying to render the message in a Server or a Client Component?

If it's a Server Component, it seems like somehow no `messages` are returned in your `i18n/request.ts` for this page? Maybe adding some logging in that file could help to diagnose. The "on the provider" part is actually misleading in this case, I'll [adjust this](https://github.com/amannn/next-intl/pull/1589) (thanks!).

Any chance the negative lookahead in your middleware somehow matches the post you're looking at? I'm typically suggesting [matching on locale prefixes](https://next-intl-docs.vercel.app/docs/routing/middleware#matcher-config) since it's usually safer.
